lib/fog/openstack/models/image_v2/images.rb

```diff
@@ -18,9 +18,9 @@ def summary(options = {})
           end
 
           def find_by_id(id)
-            image_hash = service.get_image_by_id(id).body
-            Fog::Image::OpenStack::V2::Image.new(
-                image_hash.merge(:service => service))
+            result = all.find { |image| image.id == id }
```

Comment:
Why did you switch here from getting an individual image to listing all images? The API should return something similar to your error message (`itemNotFound (404)`) in case the image with that id doesn't exist.